File: src/osx/syncmenu/test-menu.c
#include <gtk/gtk.h>

#include "sync-menu.h"

GtkWidget *open_item;
GtkWidget *copy_item;

static void
menu_item_activate_cb (GtkWidget *item,
                       gpointer   user_data)
{
  gboolean visible;
  gboolean sensitive;

  g_print ("Item activated: %s\n", (gchar *) user_data);

  g_object_get (G_OBJECT (copy_item),
                "visible", &visible,
                "sensitive", &sensitive,
                NULL);

  if (item == open_item) {
    gtk_widget_set_sensitive (copy_item, !sensitive);
    /*g_object_set (G_OBJECT (copy_item), "visible", !visible, NULL);*/
  }
}

static GtkWidget *
test_setup_menu (void)
{
  GtkWidget *menubar;
  GtkWidget *menu;
  GtkWidget *item;
  
  menubar = gtk_menu_bar_new ();

  item = gtk_menu_item_new_with_label ("File");
  gtk_menu_shell_append (GTK_MENU_SHELL (menubar), item);
  menu = gtk_menu_new ();
  gtk_menu_item_set_submenu (GTK_MENU_ITEM (item), menu);
  item = gtk_menu_item_new_with_label ("Open");
  open_item = item;
  g_signal_connect (item, "activate", G_CALLBACK (menu_item_activate_cb), "open");
  gtk_menu_shell_append (GTK_MENU_SHELL (menu), item);
  item = gtk_menu_item_new_with_label ("Quit");
  g_signal_connect (item, "activate", G_CALLBACK (menu_item_activate_cb), "quit");
  gtk_menu_shell_append (GTK_MENU_SHELL (menu), item);

  item = gtk_menu_item_new_with_label ("Edit");

  gtk_menu_shell_append (GTK_MENU_SHELL (menubar), item);
  menu = gtk_menu_new ();
  gtk_menu_item_set_submenu (GTK_MENU_ITEM (item), menu);
  item = gtk_menu_item_new_with_label ("Copy");
  copy_item = item;
  g_signal_connect (item, "activate", G_CALLBACK (menu_item_activate_cb), "copy");
  gtk_menu_shell_append (GTK_MENU_SHELL (menu), item);
  item = gtk_menu_item_new_with_label ("Paste");
  g_signal_connect (item, "activate", G_CALLBACK (menu_item_activate_cb), "paste");
  gtk_menu_shell_append (GTK_MENU_SHELL (menu), item);

  item = gtk_menu_item_new_with_label ("Help");
  gtk_menu_shell_append (GTK_MENU_SHELL (menubar), item);
  menu = gtk_menu_new ();
  gtk_menu_item_set_submenu (GTK_MENU_ITEM (item), menu);
  item = gtk_menu_item_new_with_label ("About");
  g_signal_connect (item, "activate", G_CALLBACK (menu_item_activate_cb), "about");
  gtk_menu_shell_append (GTK_MENU_SHELL (menu), item);

  return menubar;
}

int
main (int argc, char **argv)
{
  GtkWidget *window;
  GtkWidget *vbox;
  GtkWidget *menubar;

  gtk_init (&argc, &argv);

  window = gtk_window_new (GTK_WINDOW_TOPLEVEL);
  gtk_window_set_default_size (GTK_WINDOW (window), 400, 300);
  g_signal_connect (window, "destroy", G_CALLBACK (gtk_main_quit), NULL);

  vbox = gtk_vbox_new (FALSE, 0);
  gtk_container_add (GTK_CONTAINER (window), vbox);

  menubar = test_setup_menu ();
  gtk_box_pack_start (GTK_BOX (vbox), 
                      menubar,
                      FALSE, TRUE, 0);
  
  gtk_box_pack_start (GTK_BOX (vbox), 
                      gtk_label_new ("Some window content here"), 
                      TRUE, TRUE, 0);

  gtk_widget_show_all (window);

  gtk_widget_hide (menubar);

  sync_menu_takeover_menu (GTK_MENU_SHELL (menubar));

  gtk_main ();

  return 0;
}

File: src/osx/syncmenu/test-menu.py
import gtk, syncmenu


open_item = None
copy_item = None


def menu_item_activate_cb(item, user_data):
	print "Item activated: %s" % user_data

	#g_object_get (G_OBJECT (copy_item),
	#            "visible", &visible,
	#            "sensitive", &sensitive,
	#            NULL)

	#if (item == open_item) {
	#gtk_widget_set_sensitive (copy_item, !sensitive)
	#/*g_object_set (G_OBJECT (copy_item), "visible", !visible, NULL)*/


def test_setup_menu():
	global open_item, copy_item
	menubar = gtk.MenuBar()

	item = gtk.MenuItem("File")
	menubar.append(item)
	menu = gtk.Menu()
	item.set_submenu(menu)
	item = gtk.MenuItem("Open")
	open_item = item
	item.connect("activate", menu_item_activate_cb, "open")
	menu.append(item)
	item = gtk.MenuItem("Quit")
	item.connect("activate", menu_item_activate_cb, "quit")
	menu.append(item)

	item = gtk.MenuItem("Edit")

	menubar.append(item)
	menu = gtk.Menu()
	item.set_submenu(menu)
	item = gtk.MenuItem("Copy")
	copy_item = item
	item.connect("activate", menu_item_activate_cb, "copy")
	menu.append(item)
	item = gtk.MenuItem("Paste")
	item.connect("activate", menu_item_activate_cb, "paste")
	menu.append(item)

	item = gtk.MenuItem("Help")
	menubar.append(item)
	menu = gtk.Menu()
	item.set_submenu(menu)
	item = gtk.MenuItem("About")
	item.connect("activate", menu_item_activate_cb, "about")
	menu.append(item)

	return menubar


window = gtk.Window(gtk.WINDOW_TOPLEVEL)
window.set_default_size(400, 300)
window.connect("destroy", gtk.main_quit, None)

vbox = gtk.VBox(False, 0)
window.add(vbox)

menubar = test_setup_menu()
vbox.pack_start(menubar, False, True, 0)
vbox.pack_start(gtk.Label("Some window content here"), True, True, 0)

window.show_all()
menubar.hide()

syncmenu.takeover_menu(menubar)

gtk.main()

# vim: se ts=3:
